src/structures/ShardManager.ts

```diff
@@ -0,0 +1,19 @@
+import DiscordWebsocket from '../../ws/DiscordWebsocket';
+import Collection from '../util/Collection';
+
+export default class ShardManager {
+  shards: Collection<DiscordWebsocket>;
+
+  private shardID: number;
+
+  constructor() {
+    this.shards = new Collection({ base: DiscordWebsocket });
+    this.shardID = 0;
+  }
+
+  connect(websocket: DiscordWebsocket) {
+    websocket.connect();
+
+    this.shards.set((this.shardID++).toString(), websocket);
```

Comment:
Don't need to `toString()` this (if it requires a string, leave it and I'll fix it later)